platform/server/elasticsearch/__tests__/ElasticsearchConfig.test.ts

```diff
@@ -0,0 +1,37 @@
+import { ElasticsearchConfig } from '../ElasticsearchConfig';
+import { ClusterSchema } from '../schema';
+
+test('filters headers', () => {
+  const clusterSchema = {} as ClusterSchema;
+  const headers = { foo: 'bar', baz: 'qux' };
+  const config = new ElasticsearchConfig('data', clusterSchema);
+  config.requestHeadersWhitelist = ['foo'];
+
+  const expectedHeaders = { foo: 'bar' };
+  expect(config.filterHeaders(headers)).toEqual(expectedHeaders);
+});
+
+describe('creates elasticsearch client config', () => {
+  test('when shouldAuth is true by default (for data clients)', () => {
+    const clusterSchema = {
+      pingTimeout: { asMilliseconds: () => {} },
+      requestTimeout: { asMilliseconds: () => {} },
+      url: '',
+      username: 'foo',
+      password: 'bar'
+    } as ClusterSchema;
+    const config = new ElasticsearchConfig('data', clusterSchema);
+
+    expect(config.toElasticsearchClientConfig()).toMatchSnapshot();
+  });
+
+  test('when shouldAuth is false (for admin clients)', () => {});
```

Comment:
__typo:__ looks like a typo, `test` doesn't do anything and we have "independent" `expect` :)

Reply:
yikes. good catch. this is definitely not good!